ClientApp/src/app/title-inplace/title-inplace.component.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  ViewChild,
+} from '@angular/core';
+
+import { MessageService } from 'primeng/api';
+import { Inplace } from 'primeng/inplace';
+
+import { ProjectData } from '../../types/project';
+import { TaskData } from '../../types/task';
+
+@Component({
+  selector: 'title-inplace',
+  templateUrl: './title-inplace.component.html',
+  styleUrls: ['./title-inplace.component.scss'],
+  providers: [MessageService],
+})
+export class TitleInplaceComponent implements OnInit {
+  @Input() entityData!: TaskData | ProjectData;
+
+  @ViewChild('titleInplace') titleInplace!: Inplace;
+
+  updatedTitle!: string;
+
+  constructor(
+    private messageService: MessageService,
+  ) { }
+
+  ngOnInit() {
+    this.updatedTitle = this.entityData.name;
+  }
+
+  approveChanges() {
```

Comment:
Do you think it would be possible to have these functions in a base component that any inplace components could extend?

Reply:
The approveChanges() and cancelInput() functions I mean. I didn't place this comment that well

Reply:
Ahhhh, I was thinking about this for a while but now I see the vision. I'll separate those out this afternoon/evening and update the pr. Good pont!

Reply:
we chatted in discord and decided to not go with this idea for the time being, it poses some problems and the need isn't that pressing. the door is definitely still open if we need something like this in the future.